library/L1_Peripheral/lpc17xx/system_controller.hpp

```diff
@@ -117,273 +53,563 @@ class SystemController final : public sjsu::SystemController
+      ///
+      /// @note This value is only used if the PLL0 output is used to drive the
+      ///       USB subsystem.
+      UsbClockDivider divider = UsbClockDivider::kDivideBy6;
+    } usb = {};
+  };
+
+  /// Fixed Internal RC frequency.
+  static constexpr units::frequency::hertz_t kIrcFrequency = 4_MHz;
+
+  /// Fixed RTC frequency.
   static constexpr units::frequency::hertz_t kRTCFrequency = 32'768_Hz;
 
-  /// Pointer to the system controller peripheral in memory.
+  /// The required frequency for the USB clock.
+  static constexpr units::frequency::hertz_t kUsbClockFrequency = 48_MHz;
+
+  /// Pointer reference to the system controller peripheral in memory.
   inline static LPC_SC_TypeDef * system_controller = LPC_SC;
 
-  // TODO(#1140): Migrate to SystemController V2.0
+  /// @param clock_configuration The desired clock configurations for the
+  ///                            system.
+  explicit constexpr SystemController(
+      ClockConfiguration_t & clock_configuration)
+      : clock_configuration_(clock_configuration)
+  {
+  }
+
   void Initialize() override
   {
-    return;
+    units::frequency::hertz_t osc_clk =
+        clock_configuration_.main_oscillator.frequency;
+
+    units::frequency::hertz_t sys_clk = 0_MHz;
+    units::frequency::hertz_t pll_clk = 0_MHz;
+    units::frequency::hertz_t usb_clk = 0_MHz;
+    uint8_t cpu_clock_divider         = clock_configuration_.cpu.divider;
+
+    // =========================================================================
+    // Step 1. Enable the main oscillator if it is used to drive the system
+    //         clock and/or PLL1.
+    // =========================================================================
+    if ((clock_configuration_.usb.clock_source == UsbClockSource::kPll1) ||
+        (clock_configuration_.system.clock_source == Oscillator::kMain))
+    {
+      EnableMainOscillator();
+    }
+
+    // =========================================================================
+    // Step 2. Determine the running system clock rate based on the selected
+    //         oscillator.
+    // =========================================================================
+    switch (clock_configuration_.system.clock_source)
+    {
+      case Oscillator::kIrc: sys_clk = kIrcFrequency; break;
+      case Oscillator::kMain: sys_clk = osc_clk; break;
+      case Oscillator::kRtc: sys_clk = kRTCFrequency; break;
+    }
+
+    // =========================================================================
+    // Step 3. Configure the CPU clock by configuring and enabling PLL0 if
+    //         necessary.
+    // =========================================================================
+    switch (clock_configuration_.cpu.clock_source)
+    {
+      case SystemController::CpuClockSource::kSystemClock:
+      {
+        SetSystemClockSource(clock_configuration_.system.clock_source);
+        SetCpuClockDivider(cpu_clock_divider);
+        pll_clk = sys_clk;
+      }
+      break;
```

Comment:
```suggestion
        pll_clk = sys_clk;
        break;
      }
```
Break should be within the case scope.